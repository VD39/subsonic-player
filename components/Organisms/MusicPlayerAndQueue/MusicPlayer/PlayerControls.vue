<script setup lang="ts">
import MainPlayerControls from '@/components/Organisms/MusicPlayerAndQueue/Controls/MainPlayerControls.vue';
import RepeatButton from '@/components/Organisms/MusicPlayerAndQueue/Controls/RepeatButton.vue';
import ShuffleButton from '@/components/Organisms/MusicPlayerAndQueue/Controls/ShuffleButton.vue';

const { isRadioStation } = useAudioPlayer();
</script>

<template>
  <div :class="['centerItems', $style.playerControls]">
    <ShuffleButton v-if="!isRadioStation" />

    <MainPlayerControls />

    <RepeatButton v-if="!isRadioStation" />
  </div>
</template>

<style module>
.playerControls {
  gap: var(--default-space);
}
</style>
