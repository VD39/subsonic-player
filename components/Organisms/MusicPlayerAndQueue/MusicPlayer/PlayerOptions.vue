<script setup lang="ts">
import FavouriteButton from '@/components/Molecules/FavouriteButton.vue';
import MediaInformation from '@/components/Organisms/MusicPlayerAndQueue/Controls/MediaInformation.vue';
import PlaybackRateButton from '@/components/Organisms/MusicPlayerAndQueue/Controls/PlaybackRateButton.vue';
import QueueButton from '@/components/Organisms/MusicPlayerAndQueue/Controls/QueueButton.vue';
import VolumeControl from '@/components/Organisms/MusicPlayerAndQueue/Controls/VolumeControl.vue';

const {
  currentTime,
  currentTrack,
  duration,
  isPodcastEpisode,
  isRadioStation,
} = useAudioPlayer();
</script>

<template>
  <div :class="['centerAll', $style.playerOptions]">
    <p
      v-if="!isRadioStation && duration"
      ref="timeProgress"
      :class="['smallFont', $style.time]"
    >
      {{ secondsToHHMMSS(currentTime) }} / {{ secondsToHHMMSS(duration) }}
    </p>

    <VolumeControl />

    <PlaybackRateButton v-if="isPodcastEpisode" />

    <MediaInformation v-if="!isRadioStation" />

    <FavouriteButton
      v-if="'favourite' in currentTrack"
      :id="currentTrack.id"
      :favourite="currentTrack.favourite"
      :type="currentTrack.type"
    />

    <QueueButton />
  </div>
</template>

<style module>
.playerOptions {
  gap: var(--default-space);
  padding-right: var(--default-space);
}

.time {
  flex-shrink: 0;
  margin-right: var(--default-space);
}
</style>
