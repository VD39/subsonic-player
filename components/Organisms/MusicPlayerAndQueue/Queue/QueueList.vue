<script setup lang="ts">
import ButtonLink from '@/components/Atoms/ButtonLink.vue';
import QueueList from '@/components/Organisms/QueueList.vue';

const {
  clearQueueList,
  playTrackFromQueueList,
  queueList,
  removeTrackFromQueueList,
} = useAudioPlayer();
const { toggleQueueList } = useQueue();
</script>

<template>
  <div class="queueWrapper">
    <div class="centerAll">
      <ButtonLink
        ref="closeQueueList"
        :icon="ICONS.queueClose"
        icon-weight="bold"
        title="Close queue list"
        full-width
        class="queueAction"
        @click="toggleQueueList"
      >
        Close queue list
      </ButtonLink>
    </div>

    <div class="spaceBetween">
      <div />

      <ButtonLink
        ref="clearQueue"
        :icon="ICONS.clear"
        title="Clear queue"
        @click="clearQueueList"
      >
        Clear queue
      </ButtonLink>
    </div>

    <QueueList
      :tracks="queueList"
      @play-track="playTrackFromQueueList"
      @remove-from-queue="removeTrackFromQueueList"
    />
  </div>
</template>
