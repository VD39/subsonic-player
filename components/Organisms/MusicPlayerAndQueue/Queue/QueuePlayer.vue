<script setup lang="ts">
import ArtistsList from '@/components/Atoms/ArtistsList.vue';
import ButtonLink from '@/components/Atoms/ButtonLink.vue';
import LinkOrText from '@/components/Atoms/LinkOrText.vue';
import MarqueeScroll from '@/components/Atoms/MarqueeScroll.vue';
import FavouriteButton from '@/components/Molecules/FavouriteButton.vue';
import PreloadImage from '@/components/Molecules/PreloadImage.vue';
import MainPlayerControls from '@/components/Organisms/MusicPlayerAndQueue/Controls/MainPlayerControls.vue';
import MediaInformation from '@/components/Organisms/MusicPlayerAndQueue/Controls/MediaInformation.vue';
import PlaybackRateButton from '@/components/Organisms/MusicPlayerAndQueue/Controls/PlaybackRateButton.vue';
import RepeatButton from '@/components/Organisms/MusicPlayerAndQueue/Controls/RepeatButton.vue';
import ShuffleButton from '@/components/Organisms/MusicPlayerAndQueue/Controls/ShuffleButton.vue';
import TrackSeeker from '@/components/Organisms/MusicPlayerAndQueue/Controls/TrackSeeker.vue';

const {
  currentTime,
  currentTrack,
  duration,
  isPodcastEpisode,
  isRadioStation,
  isTrack,
} = useAudioPlayer();
const { toggleQueueList, toggleQueuePlayer } = useQueue();
</script>

<template>
  <div class="queueWrapper column">
    <ButtonLink
      ref="closeQueueMenu"
      class="queueAction"
      full-width
      :icon="ICONS.queueClose"
      icon-weight="bold"
      title="Close queue menu"
      @click="toggleQueuePlayer"
    >
      Close queue menu
    </ButtonLink>

    <div :class="['column', $style.playerWrapper]">
      <div :class="$style.wrapper">
        <div :class="$style.inner">
          <div :class="$style.imageWrapper">
            <div :class="$style.imageInner">
              <div :class="$style.preloadImageWrapper">
                <PreloadImage
                  :class="$style.preloadImage"
                  :image="currentTrack.image"
                />
              </div>
            </div>
          </div>
        </div>

        <div :class="$style.trackDetailsWrapper">
          <div :class="['mBAllL', 'column', $style.trackDetails]">
            <div>
              <MarqueeScroll inert>
                <h2 class="strong mBXS">
                  {{ currentTrack.name }}
                </h2>
              </MarqueeScroll>

              <template v-if="isTrack">
                <MarqueeScroll
                  v-if="'album' in currentTrack && currentTrack.album"
                  ref="albumMarqueeScroll"
                >
                  <LinkOrText
                    :is-link="!!currentTrack.albumId"
                    :text="currentTrack.album"
                    :to="`/album/${currentTrack.albumId}`"
                  />
                </MarqueeScroll>

                <MarqueeScroll
                  v-if="
                    'artists' in currentTrack && currentTrack.artists.length
                  "
                  ref="artistsMarqueeScroll"
                >
                  <ArtistsList :artists="currentTrack.artists" />
                </MarqueeScroll>
              </template>

              <template v-if="isPodcastEpisode">
                <LinkOrText
                  v-if="'podcastId' in currentTrack && currentTrack.podcastId"
                  ref="podcastLinkOrText"
                  :is-link="!!currentTrack.podcastId"
                  :text="currentTrack.podcastName"
                  :to="`/podcast/all/${currentTrack.podcastId}`"
                />

                <MarqueeScroll
                  v-if="'author' in currentTrack && currentTrack.author"
                  ref="authorMarqueeScroll"
                >
                  <p>{{ currentTrack.author }}</p>
                </MarqueeScroll>
              </template>
            </div>

            <div class="fullWidth">
              <TrackSeeker class="fullWidth" />

              <div
                v-if="!isRadioStation && duration"
                ref="timeProgress"
                class="smallFont centerItems spaceBetween"
              >
                <time>{{ secondsToHHMMSS(currentTime) }}</time>
                <time>{{ secondsToHHMMSS(duration) }}</time>
              </div>
            </div>

            <MainPlayerControls
              :class="['fullWidth', $style.secondaryOptions]"
            />

            <div
              :class="['centerAll', 'spaceBetween', $style.secondaryOptions]"
            >
              <FavouriteButton
                v-if="'favourite' in currentTrack"
                :id="currentTrack.id"
                :favourite="currentTrack.favourite"
                :type="currentTrack.type"
              />

              <ShuffleButton v-if="!isRadioStation" />

              <RepeatButton v-if="!isRadioStation" />

              <PlaybackRateButton v-if="isPodcastEpisode" />

              <MediaInformation v-if="!isRadioStation" />
            </div>
          </div>
        </div>
      </div>
    </div>

    <ButtonLink
      ref="openQueueList"
      class="queueAction"
      full-width
      :icon="ICONS.playlist"
      icon-weight="bold"
      title="Open queue list"
      @click="toggleQueueList"
    >
      Open queue list
    </ButtonLink>
  </div>
</template>

<style module>
.playerWrapper {
  flex: 1;
  width: var(--width-height-100);
  max-width: 700px;
  margin: auto;
}

.wrapper {
  display: flex;
  flex-direction: column;
  height: calc(100vh - (var(--space-40) * 2) - (var(--space-16) * 2));
  padding: var(--space-40) var(--space-40) var(--space-16);
}

.inner {
  margin-bottom: var(--space-40);
  overflow: hidden;
}

.imageWrapper {
  display: flex;
  max-width: 500px;
  aspect-ratio: 1;
  margin: auto;
}

.imageInner {
  position: relative;
  aspect-ratio: 1;
}

.preloadImageWrapper {
  position: absolute;
  top: 0;
  width: var(--width-height-100);
}

.imageWrapper,
.imageInner,
.preloadImageWrapper,
.preloadImage {
  height: var(--width-height-100);
}

.trackDetailsWrapper {
  display: contents;
}

.trackDetails {
  flex: 1;
  justify-content: end;
  margin-top: auto;
}

.secondaryOptions {
  max-width: 400px;
  margin: 0 auto;
}
</style>
