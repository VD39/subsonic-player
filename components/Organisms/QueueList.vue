<script setup lang="ts">
import ArtistsList from '@/components/Atoms/ArtistsList.vue';
import ButtonLink from '@/components/Atoms/ButtonLink.vue';
import LinkOrText from '@/components/Atoms/LinkOrText.vue';
import MarqueeScroll from '@/components/Atoms/MarqueeScroll.vue';
import NoMediaMessage from '@/components/Atoms/NoMediaMessage.vue';
import FavouriteButton from '@/components/Molecules/FavouriteButton.vue';
import TrackMeta from '@/components/Molecules/TrackMeta.vue';
import TrackPlayPause from '@/components/Organisms/TrackPlayPause.vue';

defineProps<{
  tracks: QueueTrack[];
}>();

defineEmits(['addToPlaylist', 'playTrack', 'removeFromQueue']);
</script>

<template>
  <template v-if="tracks.length">
    <h4>Tracks: {{ tracks.length }}</h4>

    <div ref="tracksWrapper" class="trackTable">
      <div class="trackHeader">
        <div class="trackCell">Track</div>
        <div class="trackCell trackSecondary">Album/Podcast</div>
        <div class="trackCell trackSecondary">Artists/Author</div>
        <div class="trackCell trackTime">Time</div>
        <div class="trackCell trackOptions" />
      </div>

      <div
        v-for="(track, index) in tracks"
        :key="track.id"
        class="trackRow"
        data-test-id="track"
      >
        <div class="trackCell">
          <div>
            <TrackPlayPause
              :track-id="track.id"
              :image="track.image"
              :track-number="DEFAULT_VALUE"
              @play-track="$emit('playTrack', index)"
            />

            <TrackMeta :track="track" class="trackMeta" />

            <FavouriteButton
              v-if="'favourite' in track"
              :id="track.id"
              :type="track.type"
              :favourite="track.favourite"
            />
          </div>
        </div>

        <div class="trackCell trackSecondary">
          <MarqueeScroll
            v-if="'album' in track && track.album"
            ref="albumMarqueeScroll"
          >
            <LinkOrText
              :is-link="!!track.albumId"
              :text="track.album"
              :to="`/album/${track.albumId}`"
            />
          </MarqueeScroll>

          <LinkOrText
            v-else-if="'podcastName' in track && track.podcastName"
            ref="podcastNameLinkOrText"
            :is-link="!!track.podcastId"
            :text="track.podcastName"
            :to="`/podcast/all/${track.podcastId}`"
          />

          <p v-else ref="albumElse">{{ DEFAULT_VALUE }}</p>
        </div>

        <div class="trackCell trackSecondary">
          <MarqueeScroll
            v-if="'artists' in track && track.artists.length"
            ref="artistsMarqueeScroll"
          >
            <ArtistsList :artists="track.artists" />
          </MarqueeScroll>

          <MarqueeScroll
            v-else-if="'author' in track && track.author"
            ref="authorMarqueeScroll"
          >
            <p>{{ track.author }}</p>
          </MarqueeScroll>

          <p v-else ref="artistsElse">{{ DEFAULT_VALUE }}</p>
        </div>

        <time class="trackCell trackTime">
          {{ track.duration }}
        </time>

        <div class="trackCell trackOptions">
          <ButtonLink
            ref="removeFromQueue"
            icon="PhX"
            title="Remove item from queue"
            icon-weight="bold"
            icon-size="small"
            @click="$emit('removeFromQueue', track.id)"
          >
            Remove item from queue
          </ButtonLink>
        </div>
      </div>
    </div>
  </template>

  <NoMediaMessage
    v-else
    :icon="IMAGE_DEFAULT_BY_TYPE.track"
    message="No tracks found."
  />
</template>
