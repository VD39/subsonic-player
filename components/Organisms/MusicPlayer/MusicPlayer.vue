<script setup lang="ts">
import ArtistsList from '@/components/Atoms/ArtistsList.vue';
import PlayPauseButton from '@/components/Molecules/PlayPauseButton.vue';
import PreloadImage from '@/components/Molecules/PreloadImage.vue';
import ImageLink from '@/components/Organisms/ImageLink.vue';

import TrackSeeker from './Controls/TrackSeeker.vue';
import PlayerControls from './PlayerControls.vue';
import PlayerOptions from './PlayerOptions.vue';

const { currentTrack, duration, isRadioStation, isTrack } = useAudioPlayer();
</script>

<template>
  <div :class="$style.musicPlayer">
    <TrackSeeker
      v-if="!isRadioStation && duration"
      :class="['hideMobile', $style.trackSeeker]"
    />

    <div :class="['spaceBetween', 'centerItems', $style.inner]">
      <div :class="['centerItems', $style.trackDetails]">
        <ImageLink
          v-if="isTrack && 'albumId' in currentTrack && currentTrack.albumId"
          :to="`/album/${currentTrack.albumId}`"
          :title="currentTrack.name"
          :image="currentTrack.image"
          :class="$style.image"
        />

        <PreloadImage
          v-else
          :image="currentTrack.image"
          :class="$style.image"
        />

        <div>
          <p class="strong clamp mBXS">
            {{ currentTrack.name }}
          </p>

          <ArtistsList
            v-if="'artists' in currentTrack && currentTrack.artists.length"
            :artists="currentTrack.artists"
            class="smallFont clamp"
          />
        </div>
      </div>

      <PlayPauseButton class="hideDesktop" />

      <PlayerControls class="centerAll hideMobile" />

      <PlayerOptions :class="['hideMobile', $style.playerOptions]" />
    </div>
  </div>
</template>

<style module>
.musicPlayer {
  --music-player-bottom: var(--header-height);

  position: fixed;
  inset: auto 0 var(--music-player-bottom);
  z-index: 9;
  height: var(--media-player-height);
  max-height: var(--media-player-height);
  background-color: var(--background-color);
  border-top: 1px solid var(--border-color);
  box-shadow: var(--dark-box-shadow-medium);

  @media (--tablet-up) {
    --music-player-bottom: 0;
  }
}

.trackSeeker {
  position: absolute;
  z-index: 9;
  display: block;
  transform: translateY(-50%);
}

.inner {
  --inner-gap: var(--default-space);

  position: relative;
  gap: var(--inner-gap);
  padding-right: var(--default-space);

  @media (--tablet-up) {
    --inner-gap: calc(var(--default-space) * 2);

    display: grid;
    grid-template-columns: 35% auto 25%;
  }
}

.trackDetails {
  gap: var(--default-space);
}

.image {
  flex-shrink: 0;
  width: var(--media-player-height);
  height: var(--media-player-height);
}

.playerOptions {
  justify-content: end;
}

.playPauseButton,
.nextTrack {
  transform: scale(1);
}
</style>
