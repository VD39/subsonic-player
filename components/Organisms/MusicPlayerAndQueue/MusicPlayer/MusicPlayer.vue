<script setup lang="ts">
import ArtistsList from '@/components/Atoms/ArtistsList.vue';
import MarqueeScroll from '@/components/Atoms/MarqueeScroll.vue';
import PlayPauseButton from '@/components/Molecules/PlayPauseButton.vue';
import PreloadImage from '@/components/Molecules/PreloadImage.vue';
import ImageLink from '@/components/Organisms/ImageLink.vue';
import QueueButton from '@/components/Organisms/MusicPlayerAndQueue/Controls/QueueButton.vue';
import TrackSeeker from '@/components/Organisms/MusicPlayerAndQueue/Controls/TrackSeeker.vue';

import PlayerControls from './PlayerControls.vue';
import PlayerOptions from './PlayerOptions.vue';

const { currentTrack } = useAudioPlayer();
</script>

<template>
  <div :class="$style.musicPlayer">
    <TrackSeeker :class="['desktopOnly', $style.trackSeeker]" />

    <div :class="['spaceBetween', 'centerItems', $style.inner]">
      <div :class="['centerItems', $style.trackDetailsWrapper]">
        <QueueButton :class="['mobileOnly', $style.queueControl]" />

        <ImageLink
          v-if="'albumId' in currentTrack && currentTrack.albumId"
          ref="albumImageLink"
          :class="$style.image"
          :image="currentTrack.image"
          :title="`Go to album ${currentTrack.name}`"
          :to="`/album/${currentTrack.albumId}`"
        />

        <ImageLink
          v-else-if="'podcastId' in currentTrack && currentTrack.podcastId"
          ref="podcastImageLink"
          :class="$style.image"
          :image="currentTrack.image"
          :title="`Go to podcast ${currentTrack.podcastName}`"
          :to="`/podcast/all/${currentTrack.podcastId}`"
        />

        <PreloadImage
          v-else
          :class="$style.image"
          :image="currentTrack.image"
        />

        <div :class="$style.trackDetails">
          <MarqueeScroll>
            <p class="strong mBXS">
              {{ currentTrack.name }}
            </p>
          </MarqueeScroll>

          <MarqueeScroll
            v-if="'artists' in currentTrack && currentTrack.artists.length"
            ref="artistsMarqueeScroll"
          >
            <ArtistsList :artists="currentTrack.artists" class="smallFont" />
          </MarqueeScroll>

          <MarqueeScroll
            v-if="'author' in currentTrack && currentTrack.author"
            ref="authorMarqueeScroll"
          >
            <p>{{ currentTrack.author }}</p>
          </MarqueeScroll>
        </div>
      </div>

      <PlayPauseButton :class="['mobileOnly', $style.playPauseButton]" />

      <PlayerControls class="centerAll desktopOnly" />

      <PlayerOptions :class="['desktopOnly', $style.playerOptions]" />
    </div>
  </div>
</template>

<style module>
.musicPlayer {
  --music-player-bottom: var(--header-height);

  position: fixed;
  inset: auto 0 var(--music-player-bottom);
  z-index: 9;
  height: var(--media-player-height);
  max-height: var(--media-player-height);
  background-color: var(--background-color);
  border-top: 1px solid var(--border-color);
  box-shadow: var(--dark-box-shadow-medium);

  @media (--tablet-up) {
    --music-player-bottom: 0;
  }
}

.trackSeeker {
  position: absolute;
  z-index: 9;
  display: block;
  transform: translateY(-50%);
}

.inner {
  position: relative;

  @media (--tablet-up) {
    display: grid;
    grid-template-columns: 25% auto 25%;
  }
}

.trackDetailsWrapper {
  position: relative;
  gap: var(--default-space);
}

.trackDetails,
.trackDetailsWrapper {
  flex: 1;
  overflow: hidden;
}

.queueControl {
  position: absolute;
  inset: 0;
  z-index: 1;
  opacity: 0;
}

.image,
.playPauseButton {
  flex-shrink: 0;
  width: var(--media-player-height);
  height: var(--media-player-height);
}

.playPauseButton {
  justify-content: center;
}

.playerOptions {
  justify-content: end;
}
</style>
