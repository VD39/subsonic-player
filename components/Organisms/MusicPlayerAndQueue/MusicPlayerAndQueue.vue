<script setup lang="ts">
import MusicPlayer from '@/components/Organisms/MusicPlayerAndQueue/MusicPlayer/MusicPlayer.vue';
import QueueList from '@/components/Organisms/MusicPlayerAndQueue/Queue/QueueList.vue';
import QueuePlayer from '@/components/Organisms/MusicPlayerAndQueue/Queue/QueuePlayer.vue';

const { showMediaPlayer } = useAudioPlayer();
const { isQueueListOpened, isQueuePlayerOpened } = useQueue();
</script>

<template>
  <transition name="slide-up-down">
    <MusicPlayer v-if="showMediaPlayer" />
  </transition>

  <transition name="slide-up-down">
    <QueuePlayer v-if="isQueuePlayerOpened" />
  </transition>

  <transition name="slide-up-down">
    <QueueList v-if="isQueueListOpened" />
  </transition>
</template>
