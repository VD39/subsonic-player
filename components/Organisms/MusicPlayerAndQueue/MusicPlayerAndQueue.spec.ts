import type { VueWrapper } from '@vue/test-utils';

import MusicPlayer from '@/components/Organisms/MusicPlayerAndQueue/MusicPlayer/MusicPlayer.vue';
import QueueList from '@/components/Organisms/MusicPlayerAndQueue/Queue/QueueList.vue';
import QueuePlayer from '@/components/Organisms/MusicPlayerAndQueue/Queue/QueuePlayer.vue';
import { useAudioPlayerMock } from '@/test/useAudioPlayerMock';
import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { mount } from '@vue/test-utils';

import MusicPlayerAndQueue from './MusicPlayerAndQueue.vue';

const queueListOpenedMock = ref(false);
const queuePlayerOpenedMock = ref(false);

mockNuxtImport('useQueue', () => () => ({
  queueListOpened: queueListOpenedMock,
  queuePlayerOpened: queuePlayerOpenedMock,
}));

const { showMediaPlayerMock } = useAudioPlayerMock();

function factory(props = {}) {
  return mount(MusicPlayerAndQueue, {
    attachTo: document.body,
    global: {
      stubs: {
        MusicPlayer: true,
        QueueList: true,
        QueuePlayer: true,
      },
    },
    props: {
      ...props,
    },
  });
}

describe('MusicPlayerAndQueue', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = factory();
  });

  describe('when showMediaPlayer value is false', () => {
    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('does not show the MusicPlayer component', () => {
      expect(wrapper.findComponent(MusicPlayer).exists()).toBe(false);
    });
  });

  describe('when showMediaPlayer value is true', () => {
    beforeEach(() => {
      showMediaPlayerMock.value = true;
    });

    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('shows the MusicPlayer component', () => {
      expect(wrapper.findComponent(MusicPlayer).exists()).toBe(true);
    });
  });

  describe('when queuePlayerOpened is false', () => {
    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('does not show the QueuePlayer component', () => {
      expect(wrapper.findComponent(QueuePlayer).exists()).toBe(false);
    });
  });

  describe('when queuePlayerOpened is true', () => {
    beforeEach(() => {
      queuePlayerOpenedMock.value = true;
    });

    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('shows the QueuePlayer component', () => {
      expect(wrapper.findComponent(QueuePlayer).exists()).toBe(true);
    });
  });

  describe('when queueListOpened is false', () => {
    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('does not show the QueueList component', () => {
      expect(wrapper.findComponent(QueueList).exists()).toBe(false);
    });
  });

  describe('when queueListOpened is true', () => {
    beforeEach(() => {
      queueListOpenedMock.value = true;
    });

    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('shows the QueueList component', () => {
      expect(wrapper.findComponent(QueueList).exists()).toBe(true);
    });
  });
});
