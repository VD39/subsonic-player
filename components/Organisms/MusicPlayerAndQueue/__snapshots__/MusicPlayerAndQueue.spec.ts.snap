// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MusicPlayerAndQueue > when queueListOpened is false > matches the snapshot 1`] = `
"<music-player-stub></music-player-stub>
<queue-player-stub></queue-player-stub>
<!--v-if-->"
`;

exports[`MusicPlayerAndQueue > when queueListOpened is true > matches the snapshot 1`] = `
"<music-player-stub></music-player-stub>
<queue-player-stub></queue-player-stub>
<queue-list-stub class="slide-up-down-enter-from slide-up-down-enter-active"></queue-list-stub>"
`;

exports[`MusicPlayerAndQueue > when queuePlayerOpened is false > matches the snapshot 1`] = `
"<music-player-stub></music-player-stub>
<!--v-if-->
<!--v-if-->"
`;

exports[`MusicPlayerAndQueue > when queuePlayerOpened is true > matches the snapshot 1`] = `
"<music-player-stub></music-player-stub>
<queue-player-stub class="slide-up-down-enter-from slide-up-down-enter-active"></queue-player-stub>
<!--v-if-->"
`;

exports[`MusicPlayerAndQueue > when showMediaPlayer value is false > matches the snapshot 1`] = `
"<!--v-if-->
<!--v-if-->
<!--v-if-->"
`;

exports[`MusicPlayerAndQueue > when showMediaPlayer value is true > matches the snapshot 1`] = `
"<music-player-stub class="slide-up-down-enter-from slide-up-down-enter-active"></music-player-stub>
<!--v-if-->
<!--v-if-->"
`;
